enterprise/backend/src/metabase_enterprise/semantic_search/index_metadata.clj

```diff
@@ -62,18 +70,22 @@
   the same database."
   [index index-metadata]
   ;; if created databases for namespacing in tests we could probably remove this whole idea.
-  (let [{:keys [index-table-qualifier]} index-metadata
-        qualify (partial format index-table-qualifier)]
+  (let [{:keys [index-table-qualifier index-table-unique-suffix]} index-metadata
+        qualify (fn [table-name]
+                  (cond-> (format index-table-qualifier table-name)
+                    index-table-unique-suffix
+                    (str "_" (model-table-suffix))))]
     (-> index
         (update :table-name qualify))))
 
 (def default-index-metadata
   "The default index metadata configuration that will be used for the search engine integration."
-  {:version               "1"
-   :metadata-table-name   "index_metadata"
-   :control-table-name    "index_control"
-   :gate-table-name       "index_gate"
-   :index-table-qualifier "%s"})
+  {:version                   "1"
+   :metadata-table-name       "index_metadata"
+   :control-table-name        "index_control"
+   :gate-table-name           "index_gate"
+   :index-table-qualifier     "%s"
+   :index-table-unique-suffix true})
```

Comment:
I am not sure why we need to change anything about index-metadata to accomplish this? Index metadata does not care what the table name is AFAIK - Other than the relationship to some kind of 'canonical' index value in `find-best!`, which you've already fixed.

Not a blocker - but I do think it would be simpler.

Opinion: The map / qualifier here is a short term isolation hack and pg schemas would likely be better post 'incremental migration support' if we want to take advantage of isolation. In this world, you no longer need to qualify table names at all by default, but you would still need to apply your suffix, so if that happens at the index level - we do not need to move stuff around later. OTOH this may never happen so take with 🧂

Reply:
I've moved the suffixing as part of `model-table-name` and got rid of `index-table-unique-suffix `. 
It was supposed to be a toggle for ease of testing a-la `index-table-qualifier`, but given that we're intending `index-table-qualifier` to be just a stopgap, I'm happy to avoid adding more.